---
title: "Resource Heterogeneity Structures Microbial Communities"
author: "Mario E. Muscarella"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
   - \usepackage{booktabs}
output: 
  pdf_document:
    fig_caption: true
---

# Introduction
Much is already know about how spatial gradients in resource availability contribution to the structure and fucntion of microbial communities.
However, we are begining to appreciate the molecular diversity within the resource pool.
Resources can be just as diverse as consumers and differ in their quality and availability to consumers.
As such, resource diverse represents a mechanisms to understand spatial distirubitons of consumers.
In this study, we explore how both the concentration and diversity of resources contribute to the structure and function of aquatic microbial communities. 

We explore three hypotheses:

1. Is there a relationship between resource concentration and diversity?
  + Dilution Hypothesis: Arrieta et al. 2015 Science
  
2. Do the concentrations and diversity of resources explain differences between communities?
  + Resource Heterogeneity Hypothesis
  
3. Do concentration and diversity explain different aspects of diversity
  + Grinnellian (environmental habitat) vs. Eltonian (food habitat) Niche Hypothesis
  + Resource Substitution Hypothesis

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls())
getwd()
setwd("~/GitHub/ResourceHeterogeneity/analyses")

# Import Tools and Standard Functions
source("../bin/MothurTools.R")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}

# Save Standard Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Load Required Packages
require("xtable")
require("png")
require("grid")
require("vegan")
require("RgoogleMaps")
library("maps")
library("mapdata")
require("maptools")
require("picante")
```

# Study System Map
We sampled 10 lakes in the Huron Mountains of Michigan. 
The Huron Mountains are located in the Superior Bedrock Uplands region of the Michigan Upper Peninsula (Schaetzl et al 2013).

The region is classified as Superior Bedrock Uplands
The forests around the lakes are old-growth hemlock-nothern hardwoods (Kerry Woods)
The lakes are part of the Pine River Watershed which drains into Lake Superior (see: www.co.marquette.mi.us/departments/plannings/docs/watershed.pdf)

# Lake Information
```{r}
lake.data <- read.csv("../data/lake_data.txt", row.names=1)
colnames(lake.data) <- c("depth", "pH", "DO1", "DO2")
lake.data$area <- rep(NA, dim(lake.data)[1])

# Shape file source: http://www.mcgi.state.mi.us/mgdl/?rel=ext&action=sext
shape <- readShapePoly("../data/lake_polygons_200403/lake_polygons_200403.shp", 
                       IDvar = "UNIQUE_ID")

# Extract Areas
grp.opt <- shape$COUNTY == "Marquette" & shape$FMU == "LSW"
lake.data$area[1] <- shape$AREA[which(shape$NAME == "Lake Ann" & grp.opt)]
lake.data$area[2] <- shape$AREA[which(shape$NAME == "Canyon Lake" & grp.opt)]
lake.data$area[3] <- shape$AREA[which(shape$NAME == "Howe Lake" & grp.opt)]
lake.data$area[4] <- shape$AREA[which(shape$NAME == "Ives Lake" & grp.opt)]
lake.data$area[5] <- shape$AREA[which(shape$NAME == "Lily Pond" & grp.opt)]
lake.data$area[6] <- shape$AREA[which(shape$NAME == "Mountain Lake" & grp.opt)]
lake.data$area[7] <- 2000 # estimate...smallest and not in GIS dataset
lake.data$area[8] <- shape$AREA[which(shape$NAME == "Rush Lake" & grp.opt)]
lake.data$area[9] <- shape$AREA[which(shape$NAME == "Second Lake" & grp.opt)]
lake.data$area[10] <- shape$AREA[which(shape$NAME == "Third Lake" & grp.opt)]

# Convert sq m to sq km
lake.data$area <- lake.data$area / 1000000

# Notes on how to get shapefile info
# str(shape, max.level=4)
# Attempt to find Pony
# plot(shape[which(shape$LAKE_NAME == "no name" & 
# shape$COUNTY == "Marquette" & shape$FMU == "LSW"), ])

# Plot from shapefile (just to check)
#  grp.opts <- shape$COUNTY == "Marquette" & shape$FMU == "LSW"
#  plot(shape[c(
#   which(shape$NAME == "Ives Lake" & grp.opts),
#   which(shape$NAME == "Mountain Lake" & grp.opts),
#   which(shape$NAME == "Howe Lake" & grp.opts),
#   which(shape$NAME == "Rush Lake" & grp.opts),
#   which(shape$NAME == "Lake Ann" & grp.opts),
#   which(shape$NAME == "Canyon Lake" & grp.opts),
#   which(shape$NAME == "Lily Pond" & grp.opts),
#   which(shape$NAME == "Second Lake" & grp.opts),
#   which(shape$NAME == "Third Lake" & grp.opts)),])

lake.data <- lake.data[sort(row.names(lake.data)), c("area", "pH")]
```

## Supplemental Table 1: Lake Physical Properties
### Note: depth represents the maximum depth recorded during sampling and thus does not reflect the true depth of the lake

```{r, results="asis"}
addtorow <- list()
addtorow$pos <- list(0)
addtorow$command <- c("Lake & Area (Km$^2$)  & pH \\\\\n")
lake.tab <- xtable(lake.data, digits = c(0,1,3))
align(lake.tab) <- "ccc"
print(lake.tab, add.to.row = addtorow, include.colnames = FALSE, 
      type= "latex", file="../tables/SuppTable1.tex", 
      hline.after = c(-1, -1, 0, nrow(lake.tab)))
print(lake.tab, add.to.row = addtorow, include.colnames = FALSE, 
      comment = FALSE, hline.after = c(-1, -1, 0, nrow(lake.tab)))
```

## Supplemental Figure 1: Study System Map
```{r, results='hide'}
png(filename="../figures/Supp1.png",
    width = 1800, height = 900, res = 96*2)
par(opar)
par(mfrow = c(1,1), mar = c(0, 0, 0, 0), oma = c(0, 0, 0, 0) + 0.5)

newmap1 <- GetMap(center = c(46.86, -87.93), zoom = 13, 
                 maptype = "terrain", GRAYSCALE = FALSE, frame = FALSE,
                 path = "&style=feature:all|element:labels|visibility:off")
newmap2 <- GetMap(center = c(46.86, -87.82), zoom = 13, 
                 maptype = "terrain", GRAYSCALE = FALSE, frame = FALSE,
                 path = "&style=feature:all|element:labels|visibility:off")

layout( matrix(c(1,1,1,1,1,1,
                 1,1,1,1,1,1,
                 1,1,1,1,1,1,
                 2,2,2,2,2,2,
                 2,2,2,3,3,3,
                 2,2,2,3,3,3), 6, 6, byrow = FALSE),
        widths = rep(2, 6), heights = rep(2, 6))

# Left Side
PlotOnStaticMap(newmap1, zoom = 13, cex = 2, col = "blue")
text(-120, 270, "Howe", col="red", cex=1.2)
text(100, 255, "Rush", col="red", cex=1.2)
text(105, 55, "Mountain", col="red", cex=1.2)
arrows(10, 100, x1 = -10, y1 = 100, length=0.05, col = "red", lwd = 2, code = 1)
text(-36, 100, "Ann", col="red", cex=1.2)
arrows(60, 230, x1 = 40, y1 = 230, length=0.05, col = "red", lwd = 2, code = 1)
text(8, 230, "Pony", col="red", cex=1.2)
arrows(55, -230, x1 = 75, y1 = -230, length=0.05, col = "red", lwd = 2, code = 1)
text(120, -230, "Canyon", col="red", cex=1.2)

# Right Side
PlotOnStaticMap(newmap2, zoom = 13, cex = 2, col = "blue")
text(-180, -90, "Ives", col="red", cex=1.2)
arrows(-160, 20, x1 = -140, y1 = 20, length = 0.05, col = "red", lwd = 2, code = 1)
text(-75, 20, "Upper Pine", col="red", cex = 1.2)
arrows(-220, 80, x1 = -200, y1 = 80, length = 0.05, col = "red", lwd = 2, code = 1)
text(-125, 80, "Second Pine", col="red", cex = 1.2)
arrows(-55, -105, x1 = -35, y1 = -105, length = 0.05, col = "red", lwd = 2, code = 1)
text(-12, -105, "Lily", col="red", cex=1.2)

# Compass Arrow
arrows(280, 280, 280, 240, length = 0.1, col = "black", lwd = 3, code = 1)
text(280, 220, "N", col = "black", cex = 1.5)

# Inset
map("state", "Michigan", col = "gray80", fill = TRUE, 
    xlim = c(-92,-82), ylim = c(41, 48))
points(-87.89, 46.8, pch = 20, col = "red")

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices 
par(opar)
```

```{r fig.width=7, fig.height=3.5,echo=FALSE,fig.cap="Study System Map"}
img <- readPNG("../figures/Supp1.png")
grid.raster(img)
```

This study system is know to have differences in the concentrations of growth limiting nutrients.
Specifically, the concentrations of dissolved organic carbon and phosphorus differ between the lakes.
We also have data for total nitrogen, but the values are odd and may not be reasonable for interpretation.
The reason the nitrogen values are odd is because many were at the lower detection level of the instrument.

# Lake Nutrient Concentrations
```{r}
# DOC
DOC2011 <- read.delim("../data/2011DOC_data.txt", header=T)
DOC2012 <- read.delim("../data/2012DOC_data.txt", header=T)
DOC <- rbind(DOC2011, DOC2012)
DOC <- DOC[grep("MEM*", DOC$Sample), ]
colnames(DOC) <- c("sample", "conc", "LCL", "UCL", "se")
DOCkey <- read.delim("../data/DOC_KEY_epi.txt", header=T)
DOC$code <- DOC$sample
DOC <- DOC[which(DOC$code %in% DOCkey$Sample.Name), ]
DOC$sample <- DOCkey$Site[match(DOCkey$Sample.Name, DOC$code)]
DOC$year <- substr(DOC$code, 4, 7)
DOC$conc <- pmax(DOC$conc, 0)
DOC2 <- data.frame("sample" = DOC$sample, "year" = DOC$year, 
                   "conc" =  DOC$conc)[order(DOC$sample, DOC$year), ]
DOC$sample[grep("Pony", DOC$sample)] <- "Pony"
DOC <- droplevels(DOC)

# Total Nitrogen
#TN2011 <- read.delim("../data/2011TN_data.txt", header=T)
#TN2012 <- read.delim("../data/2012TN_data.txt", header=T)
TN <- read.delim("../data/HMWF_TN.txt")
#TN <- TN[grep("MEM*", TN$Sample), ]
colnames(TN) <- c("sample", "year", "conc")
#TNkey <- read.delim("../data/DOC_KEY_epi.txt", header=T)
#TN$code <- TN$sample
#TN <- TN[which(TN$code %in% TNkey$Sample.Name), ]
#TN$sample <- TNkey$Site[match(TNkey$Sample.Name, TN$code)]
#TN$year <- substr(TN$code, 4, 7)
#TN$conc <- pmax(TN$conc, 0)
TN2 <- data.frame("sample" = TN$sample, "year" = TN$year, 
                  "conc" = TN$conc)[order(TN$sample, TN$year), ]
#TN$sample[grep("Pony", TN$sample)] <- "Pony"
TN <- droplevels(TN)

# Total Phosphorus
TP2011 <- read.delim("../data/2011TP_data.txt")
TP2012 <- read.delim("../data/2012TP_data.txt")
TP2011$year <- rep("2011", dim(TP2011)[1])
TP2012$year <- rep("2012", dim(TP2012)[1])
TP <- rbind(TP2011, TP2012)
TP <- TP[grep("*iltered", TP$Sample), ]
colnames(TP) <- c("sample", "conc", "LCL", "UCL", "se", "year")
TP$code <- TP$sample
TDP <- TP[grep("*Filtered", TP$sample), ]
TP <- TP[grep("*Unfiltered", TP$sample), ]
TP$sample <- gsub(" Unfiltered", "", TP$sample)
TDP$sample <- gsub(" Filtered", "", TDP$sample)
TP[6, ] <- TDP[6, ]
TP <- TP[-c(which(TP$sample == "CanyonHypo" | TP$sample == "CanyonChemo")), ]
TP$sample <- gsub("CanyonEpi", "Canyon", TP$sample)
TP$sample <- as.factor(TP$sample)
TP$conc <- pmax(TP$conc, 0)
TP2 <- data.frame("sample" = TP$sample, "year" = TP$year, 
                  "conc" = TP$conc)[order(TP$sample, TP$year), ]
TP$sample[grep("Pony", TP$sample)] <- "Pony"
TP <- droplevels(TP)
```

## Save Data Table
```{r}
DOC2 <- aggregate(conc ~ sample + year, DOC2, mean)
TN2 <- aggregate(conc ~ sample + year, TN2, mean)
TP2 <- aggregate(conc ~ sample + year, TP2, mean)

nuts <- data.frame("sample" = DOC2$sample, "year" = DOC2$year, 
              "DOC" = DOC2$conc, "TP" = TP2$conc)

nuts <- data.frame(nuts[-which(nuts$sample == "Pony.N"), ])
nuts$sample[grep("Pony", nuts$sample)] <- "Pony"
nuts <- droplevels(nuts)
nuts$TN <- TN2$conc
```

## Table 1: Lake Nutrients
```{r, results="asis"}
nuts2 <- data.frame(matrix(NA, 10, 6))
row.names(nuts2) <- levels(nuts$sample)
colnames(nuts2) <- c("DOC11", "DOC12", "TP11", "TP12", "TN11", "TN12")
for (i in row.names(nuts2)){
  nuts2[i, 1] <- round(nuts[nuts$sample == i & nuts$year == "2011", 3], 2)
  nuts2[i, 2] <- round(nuts[nuts$sample == i & nuts$year == "2012", 3], 2)
  nuts2[i, 3] <- round(nuts[nuts$sample == i & nuts$year == "2011", 4], 2)    
  nuts2[i, 4] <- round(nuts[nuts$sample == i & nuts$year == "2012", 4], 2)
  nuts2[i, 5] <- round(nuts[nuts$sample == i & nuts$year == "2011", 5], 2)    
  nuts2[i, 6] <- round(nuts[nuts$sample == i & nuts$year == "2012", 5], 2)
}

addtorow <- list()
addtorow$pos <- list(0, 0)
addtorow$command <- c("& \\multicolumn{2}{c}{DOC (mg C L$^{-1}$)} & 
                      \\multicolumn{2}{c}{TP ($\\mu$g P L$^{-1}$)} & 
                      \\multicolumn{2}{c}{TN (mg N L$^{-1}$)}\\\\\n", 
                      "Lake & 2011 & 2012 & 2011 & 2012 & 2011 & 2012 \\\\\n")
nut.tab <- xtable(nuts2)
align(nut.tab) <- "crrrrrr"
print(nut.tab, add.to.row = addtorow, include.colnames = FALSE, 
      type= "latex", file="../tables/table1.tex", 
      hline.after = c(-1, -1, 0, nrow(nut.tab)))
print(nut.tab, add.to.row = addtorow, include.colnames = FALSE, 
      comment = FALSE, hline.after = c(-1, -1, 0, nrow(nut.tab)))
```

## Figure 1: Lake Nutrients
```{r, results='hide'}
png(filename="../figures/Figure1.png",
    width = 1600, height =1200, res = 96*2)
par(opar)
par(mfrow = c(1,1), mar = c(0, 6, 0, 0) + 0.5, oma = c(4, 0, 1, 1) + 0.5)
layout(rbind(1, 2, 3), height = c(3, 3, 3)) 

labs <- c("Ann", "Canyon", "Howe", "Ives", "Lily", "Mountain", "Pony", "Rush",
         "Second\nPine","Upper\nPine")

# DOC Plot
plot(DOC$conc ~ DOC$sample, ylim = c(0, 35), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck = -0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=1, lwd.ticks = 2, tck = 0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1.2, las = 1, 
     at = c(0, 10, 20, 30))
axis(side=2, lwd.ticks = 2, tck = 0.01, labels = F, cex.axis = 1, 
     at = c(0, 10, 20, 30))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(0, 10, 20, 30))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 10, 20, 30))
mtext(side = 2, expression(paste("DOC (mg C L" ^-1, ")", sep="")), 
      line = 3.5, cex = 1)
legend("topleft", "A", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

# Total Nitrogen Plot
plot(TN$conc ~ TN$sample, ylim = c(0,2), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1.2, las = 1, 
     at = c(0.0, 0.5, 1.0, 1.5))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0.0, 0.5, 1.0, 1.5))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(0.0, 0.5, 1.0, 1.5))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0.0, 0.5, 1.0, 1.5))
mtext(side = 2, expression(paste("TN (mg N L" ^-1, ")")), 
      line = 3.5, cex = 1)
legend("topleft", "B", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

# Total Phosphorus Plot                       
plot(TP$sample, TP$conc, ylim = c(0,50), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1.2, las = 1, 
     at = c(0, 20, 40))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 20, 40))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(0, 20, 40))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 20, 40))
mtext(side = 2, expression(paste("TP (",mu, "g P L" ^-1, ")")), 
      line = 3.5, cex = 1)
mtext(side = 1, text = labs, line = 1, at = seq(1:10), padj = 0.5, cex = 0.8)
legend("topleft", "C", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices 
par(opar)
```

```{r fig.width=6, fig.height=5,echo=FALSE,fig.cap="Lake Nutrients"}
img <- readPNG("../figures/Figure1.png")
grid.raster(img)
```

But there are other differences between these lakes. 
For example, algal derived resources differ (as infered from chlorophyll *a* concentrations).
Additionally, bacterial respiration differs between lakes though there is also temperature dependence with these values as well (temperature between years of sampling). 

# Chlorophyll a and Bacterial Respiration Ecosystem Measures
```{r}
chla <- read.delim("../data/ChlorophyllA.txt")
resp <- read.delim("../data/Respiration.txt")
# npp
# CO2 flux
# EcoResp
# DO
```

## Table 2: Chlorophyll a and BR
```{r, results="asis"}
eco <- data.frame(matrix(NA, 10, 4))
row.names(eco) <- levels(chla$Lake)
colnames(eco) <- c("Chl11", "Chl12", "Resp11", "Resp12")
for (i in row.names(eco)){
  eco[i, 1] <- round(mean(chla[chla$Lake == i & chla$Year == "2011", 3]), 2)
  eco[i, 2] <- round(mean(chla[chla$Lake == i & chla$Year == "2012", 3]), 2)
  eco[i, 3] <- round(mean(resp[resp$sample == i & resp$year == "2011", 4]), 3)    
  eco[i, 4] <- round(mean(resp[resp$sample == i & resp$year == "2012", 4]), 3)
}

addtorow <- list()
addtorow$pos <- list(0, 0, 0)
addtorow$command <- c("& \\multicolumn{2}{c}{Chl $\\emph{a}$} & 
                      \\multicolumn{2}{c}{Resp.} \\\\\n", 
                      "& \\multicolumn{2}{c}{($\\mu$g L$^{-1}$)} & 
                      \\multicolumn{2}{c}{($\\mu$M O$_2$ Hr$^{-1}$)} \\\\\n",
                      "Lake & 2011 & 2012 & 2011 & 2012 \\\\\n")
eco.tab <- xtable(eco)
align(eco.tab) <- c("c ","r ", "r ", "r ", "r ")
print(eco.tab, add.to.row = addtorow, include.colnames = FALSE, 
      type= "latex", file="../tables/table2.tex", 
      hline.after = c(-1, -1, 0, nrow(eco.tab)))
print(eco.tab, add.to.row = addtorow, include.colnames = FALSE, 
      comment = FALSE, hline.after = c(-1, -1, 0, nrow(eco.tab)))
```

## Figure 2: Ecosystem
```{r results='hide'}
chla2 <- chla[chla$Year == "2012", ]
resp2 <- resp[resp$year == "2012", ]

png(filename="../figures/Figure2.png",
    width = 1600, height =1200, res = 96*2)
par(opar)
par(mfrow = c(1,1), mar = c(0, 6, 0, 0) + 0.5, oma = c(4, 0, 1, 1) + 0.5)
layout(rbind(1, 2), height = c(4, 4)) 

labs <- c("Ann", "Canyon", "Howe", "Ives", "Lily", "Mountain", "Pony", "Rush",
         "Second\nPine","Upper\nPine")

# ChlaA Plot 2012
plot(chla2$ChlA ~ chla2$Lake, ylim = c(0, 20), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1:10))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1.2, las = 1, 
     at = c(0, 10, 20))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 10, 20, 30))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(0, 10, 20, 30))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 10, 20, 30))
mtext(side = 2, expression(paste("Chl a (mg C L" ^-1, ")", sep="")), 
      line = 3.5, cex = 1)
legend("topleft", "A", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

# Bacterial Respiration
plot(resp2$rate ~ resp2$sample, ylim = c(0.5,2), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1.2, las = 1, 
     at = c(0, 1, 2))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 1, 2))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, 
     at = c(0, 1, 2))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, 
     at = c(0, 1, 2))
mtext(side = 2, expression(paste("Resp. (", mu, "M O"[2], " Hr" ^-1, ")")), 
      line = 3.5, cex = 1)
mtext(side = 1, text = labs, line = 1, at = seq(1:10), padj = 0.5, cex = 0.8)
legend("topleft", "B", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices 
par(opar)
```

```{r fig.width=6, fig.height=5,echo=FALSE,fig.cap="Ecosystem"}
img <- readPNG("../figures/Figure2.png")
grid.raster(img)
```


# Patterns of Bacterial Diversity
The major difference that we are interested in is bacterial diversity across the sites.
We know that the lakes have different microbiomes.
These microbiomes can be influenced by physical, chemical and biological interactions within each lake.

## Import Raw Data
```{r}
# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU
design.in <- "../data/design.txt"
shared <- "../data/HMWF.bac.final.shared"
taxon  <- "../data/HMWF.bac.final.0.03.taxonomy"

# Import Design
design <- read.delim(design.in, header=T, row.names=1)

# Import Shared Files
OTUs.in <- read.otu(shared = shared, cutoff = "0.03")    # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = taxon, format = "rdp")
```

## Data Transformations
```{r}
# Reorder Site
OTUs.hmwf <- OTUs.in[rownames(design), ]

# Remove OTUs with less than two occurences across all sites
# OTUs <- OTUs.hmwf[, which(colSums(OTUs.hmwf) >= 2)]
OTUs <- OTUs.hmwf[, colSums((OTUs.hmwf > 0) * 1)  >= 2 | colSums(OTUs.hmwf >= 10)]

# Sequencing Coverage
coverage <- rowSums(OTUs)

# Good's Coverage
goods <- function(x = ""){
  1 - (sum(x == 1) / rowSums(x))
}
goods.c <- goods(OTUs)

# Make Presence Absence Matrix
OTUsPA <- (OTUs > 0) * 1

# Make Relative Abundence Matrices
OTUsREL <- OTUs
for(i in 1:dim(OTUs)[1]){
  OTUsREL[i,] <- OTUs[i,]/sum(OTUs[i,])
}

# Log Transform Relative Abundances
OTUsREL.log <- decostand(OTUs, method="log")
```

## Calculate Alpha Diversity
```{r}
# Observed Richness
S.obs <- rowSums((OTUs > 0) * 1)

# Simpson's Evenness
SimpE <- function(x = ""){
  x <- as.data.frame(x)
  D <- diversity(x, "inv")
  S <- sum((x > 0) * 1) 
  E <- (D)/S 
  return(E)
}
simpsE <- round(apply(OTUs, 1, SimpE), 3)

# Shannon's Diversity
H <- function(x = ""){
  x <- x[x>0]
  H = 0
  for (n_i in x){
    p = n_i / sum(x)
    H = H - p*log(p) 
  }
  return(H)
}

shan <- round(apply(OTUs, 1, H), 2)
shan2 <- diversity(OTUs, index = "shannon")

# Rarefied Richness
S.rar <- round(rarefy(OTUs, min(rowSums(OTUs))), 0)

design <- droplevels(design)

alpha.div <- cbind(design, S.obs, simpsE, shan, S.rar)
alpha.div <- alpha.div[order(alpha.div$Lake, alpha.div$Year, alpha.div$Molecule), ]
```

## Figure 3: Lake Alpha Diversity
```{r results='hide'}

labs <- c("Ann", "Canyon", "Howe", "Ives", "Lily", "Mountain", "Pony", "Rush",
         "Second\nPine","Upper\nPine")

png(filename="../figures/Figure3.png",
    width = 1600, height = 1100, res = 96*2)
par(opar)
par(mfrow = c(1,1), mar = c(0, 6, 0, 0) + 0.5, oma = c(3, 0, 1, 1) + 0.5)
layout(rbind(1, 2), height = c(3, 3)) 

plot(alpha.div$S.obs ~ alpha.div$Lake,  ylim = c(0,5000), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1:10))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1, las = 1, 
     at = c(1000, 3000, 5000))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1000, 3000, 5000, 7000))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1000, 3000, 5000, 7000))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1000, 3000, 5000, 7000))
mtext(side = 2, "Richness (S)", line = 4, cex = 1)
legend("topleft", "A", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

plot(alpha.div$simpsE ~ alpha.div$Lake, ylim = c(0,0.06), las = 1, 
     xaxt="n", xlab = "", yaxt="n", ylab = "")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1:10))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1:10))
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1, las = 1, 
     at = c(0, 0.02, 0.04, 0.06))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(0, 0.02, 0.04, 0.06))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1:10))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1:10))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(0, 0.02, 0.04, 0.06))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(0, 0.02, 0.04, 0.06))
mtext(side = 2, "Simpson's Eveness (E)", line = 4, cex = 1)
mtext(side = 1, text = labs, line = 0.5, at = seq(1:10), padj = 0.5, cex = 0.8)
legend("topleft", "B", bty = "n", x.intersp = 0, cex = 1.25)
box(lwd = 2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)
```

```{r fig.width=6, fig.height=4,echo=FALSE,fig.cap="Lake Nutrients"}
img <- readPNG("../figures/Figure3.png")
grid.raster(img)
```

## Alpha Diversity Statistics
```{r}
alpha.div
div.mod1 <- aov(S.obs ~ Lake, data = alpha.div)
TukeyHSD(div.mod1)
div.mod2 <- aov(simpsE ~ Lake, data = alpha.div)
TukeyHSD(div.mod2)
div.mod3 <- aov(S.rar ~ Lake, data = alpha.div)
TukeyHSD(div.mod3)
```



## Lake Phylogenetic Diversity
```{r}


data(phylocom)
phy <- phylocom$phylo
comm <- phylocom$sample
traits <- phylocom$traits

# Import Tree with *ape* 
hmwf.tree <- read.tree("../fasttree/HMWF.bac.0.03.gg.tree")
tips <- paste(rep("Otu", length(hmwf.tree$tip.label)), 
              formatC(seq(1:length(hmwf.tree$tip.label)), 
                      width = 6, format = "d", flag = "0"), sep = "")
hmwf.tree$tip.label <- tips
prunedTree <- prune.sample(OTUs,hmwf.tree)

pd(OTUs, prunedTree, include.root = FALSE)
ses.pd(OTUs, prunedTree, include.root = FALSE)
psr(OTUs, prunedTree)


require(phyloseq)
OTU.tab <- otu_table(OTUs, taxa_are_rows = F)
PHY.tree <- phy_tree(hmwf.tree)
phylo.seq <- phyloseq(OTU.tab, PHY.tree)
uni.frac.u <- UniFrac(phylo.seq, weighted = FALSE, normalized = TRUE)
uni.frac.w <- UniFrac(phylo.seq, weighted = TRUE, normalized = TRUE)
```

## Calculate and Visualize Beta Diversity
```{r}
beta.w <- function(site1 = "", site2 = ""){
  site1 = subset(site1, select = site1 > 0)               # Removes absences
  site2 = subset(site2, select = site2 > 0)               # Removes absences
  gamma = union(colnames(site1), colnames(site2))         # Gamma species pool
  s     = length(gamma)                                   # Gamma richness
  a.bar = mean(c(specnumber(site1), specnumber(site2)))   # Mean sample richness
  b.w   = round(s/a.bar - 1, 3)
  return(b.w)
}

# Calculate Bray-Curtis 
hmwf.db <- vegdist(OTUsREL, method = "bray")

# Import UniFra Distances
hmwf.uni.in <- read.table("../fasttree/HMWF.bac.0.03.gg.tree1.weighted.phylip.dist", skip=1, row.names = 1)
hmwf.uni <- as.dist(hmwf.uni.in)
```

## Principal Coordinates Analysis: Bray Curtis
```{r}
par(opar)
hmwf.pcoa <- cmdscale(hmwf.db, eig = TRUE, k = 3) 
explainvar1 <- round(hmwf.pcoa$eig[1] / sum(hmwf.pcoa$eig), 3) * 100
explainvar2 <- round(hmwf.pcoa$eig[2] / sum(hmwf.pcoa$eig), 3) * 100
explainvar3 <- round(hmwf.pcoa$eig[3] / sum(hmwf.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)
```

## Principal Coordinates Analysis: UniFrac
```{r}
par(opar)
hmwf.pcoa.b <- cmdscale(hmwf.uni, eig = TRUE, k = 3) 
explainvar1.b <- round(hmwf.pcoa.b$eig[1] / sum(hmwf.pcoa.b$eig), 3) * 100
explainvar2.b <- round(hmwf.pcoa.b$eig[2] / sum(hmwf.pcoa.b$eig), 3) * 100
explainvar3.b <- round(hmwf.pcoa.b$eig[3] / sum(hmwf.pcoa.b$eig), 3) * 100
sum.eig.b <- sum(explainvar1.b, explainvar2.b, explainvar3.b)
```

## Eigenvalue Analysis Plots
```{r}
png(filename="../figures/FigureS3.png",
    width = 1600, height = 1100, res = 96*2)
# Define Plot Parameters
layout(as.matrix(cbind(1,2)))
par(mar = c(1, 1.5, 3, 1.5) + 0.1, oma = c(5, 5, 0.5, 0.5))

# Bray Curtis Analysis
# Plot Eigenvalues
plot(hmwf.pcoa$eig, main = "Bray Curtis", 
     #xlab = "PCoA Axis", ylab = "Eigenvalue", 
     las = 1, cex.lab = 1.5, pch = 16)

# Add Expectation based on Kaiser-Guttman criterion and Broken Stick Model
abline(h = mean(hmwf.pcoa$eig), lty = 2, lwd = 2, col = "blue")
b.stick <- bstick(42, sum(hmwf.pcoa$eig))
lines(1:42, b.stick, type = "l", lty = 4, lwd = 2, col = "red")

# Add Legend
legend("topright", legend = c("Avg Eigenvalue", "Broken-Stick"), 
       lty = c(2, 4), bty = "n", col = c("blue", "red"))

axis(1, lwd=2, labels = F)
axis(2, lwd=2, labels = F)
#axis(3, lwd=2, tck=-0.01, labels = F)
#axis(4, lwd=2, tck=-0.01, labels = F)
axis(1, lwd=2, tck=0.01, labels = F)
axis(2, lwd=2, tck=0.01, labels = F)
#axis(3, lwd=2, tck=0.01, labels = F)
#axis(4, lwd=2, tck=0.01, labels = F)

box(lwd=2)

# UniFrac Analysis
# Plot Eigenvalues
plot(hmwf.pcoa.b$eig, main = "UniFrac", 
     #xlab = "PCoA Axis", ylab = "Eigenvalue", 
     las = 1, cex.lab = 1.5, pch = 16)

# Add Expectation based on Kaiser-Guttman criterion and Broken Stick Model
abline(h = mean(hmwf.pcoa.b$eig), lty = 2, lwd = 2, col = "blue")
b.stick <- bstick(42, sum(hmwf.pcoa.b$eig))
lines(1:42, b.stick, type = "l", lty = 4, lwd = 2, col = "red")

# Add Legend
legend("topright", legend = c("Avg Eigenvalue", "Broken-Stick"), 
       lty = c(2, 4), bty = "n", col = c("blue", "red"))

axis(1, lwd=2, labels = F)
axis(2, lwd=2, labels = F)
#axis(3, lwd=2, tck=-0.01, labels = F)
#axis(4, lwd=2, tck=-0.01, labels = F)
axis(1, lwd=2, tck=0.01, labels = F)
axis(2, lwd=2, tck=0.01, labels = F)
#axis(3, lwd=2, tck=0.01, labels = F)
#axis(4, lwd=2, tck=0.01, labels = F)

box(lwd=2)
mtext("PCoA Axis", side = 1, outer = T, line = 2.5, cex = 2)
mtext("Eigenvalue", side = 2, outer = T, line = 2.5, cex = 2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)
```

```{r fig.width=6, fig.height=4,echo=FALSE,fig.cap="Eigenvalue Analysis"}
img <- readPNG("../figures/FigureS3.png")
grid.raster(img)
```

## Figure 4: Bacterial Community Composition Ordination Figure: Bray Curtis
```{r results='hide'}
design$Lake_Mol <- paste(design$Lake, design$Molecule, sep = "_")

png(filename="../figures/Figure4.png",
    width = 1200, height = 1200, res = 96*2)
par(opar)
# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)

# Define Plot Symbols
lake.pch <- rep(NA, length(design$Molecule))
for (i in 1:length(design$Molecule)){
if (design$Molecule[i] == "DNA"){
    lake.pch[i] <- 16
  }else{
    lake.pch[i] <- 17
  }}


# Initiate Plot
plot(hmwf.pcoa$points[ ,1], hmwf.pcoa$points[ ,2], 
     ylim = c(-0.31, 0.25), xlim = c(-0.3, 0.6), 
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = lake.pch, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, 
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(hmwf.pcoa$points[ ,1], hmwf.pcoa$points[ ,2],
       pch = lake.pch, cex = 2.5, bg = "gray", col = "gray")

ordiellipse(cbind(hmwf.pcoa$points[ ,1], hmwf.pcoa$points[ ,2]),
    design$Lake, kind="sd", conf=0.95,
    lwd=2, draw = "polygon", col="gray", border = "black", label=TRUE, 
    cex=1, bty = 'n')

ordihull(cbind(hmwf.pcoa$points[ ,1], hmwf.pcoa$points[ ,2]),
    design$Molecule, 
    lwd=2, draw = "polygon", col="gray", border = "black", label=TRUE, 
    cex=1, bty = 'n')

# Add Labels
text(0.38, 0.13, "Lily")
text(0.56, -0.11, "Pony")
text(0.05, -0.19, "Upper Pine")
text(-0.18, -0.29, "Second Pine")
text(0, -0.05, "Mountain")
text(-0.25, 0, "Ives")
text(0.1, 0.1, "Canyon")
text(-0.25, 0.1, "Howe")
text(-0.2, 0.225, "Rush")
text(-0.05, 0.2, "Ann")

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Lake Nutrients"}
img <- readPNG("../figures/Figure4.png")
grid.raster(img)
```

## Figure 4: Bacterial Community Composition Ordination Figure: Unifrac
```{r results='hide'}
png(filename="../figures/Figure4b.png",
    width = 1200, height = 1200, res = 96*2)
par(opar)
# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)

# Define Plot Symbols
lake.pch <- rep(NA, length(design$Molecule))
for (i in 1:length(design$Molecule)){
if (design$Molecule[i] == "DNA"){
    lake.pch[i] <- 16
  }else{
    lake.pch[i] <- 17
  }}


# Initiate Plot
plot(hmwf.pcoa.b$points[ ,1], hmwf.pcoa.b$points[ ,2], 
     ylim = c(-0.25, 0.25), xlim = c(-0.35, 0.25), 
     xlab = paste("PCoA 1 (", explainvar1.b, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2.b, "%)", sep = ""),
     pch = lake.pch, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, 
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(hmwf.pcoa.b$points[ ,1], hmwf.pcoa.b$points[ ,2],
       pch = lake.pch, cex = 2.5, bg = "gray", col = "gray")

ordiellipse(cbind(hmwf.pcoa.b$points[ ,1], hmwf.pcoa.b$points[ ,2]),
    design$Molecule, kind="sd", conf=0.95,
    lwd=2, draw = "polygon", col="gray", border = "black", label=TRUE, 
    cex=1, bty = 'n')

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="UniFrac Distances"}
img <- readPNG("../figures/Figure4b.png")
grid.raster(img)
```


## Statistical Analyses
What are the differences between lakes and does resource concentration explain differences

```{r}
nuts
nuts2 <- data.frame(nuts[rep(seq_len(nrow(nuts)), each=2),])
nuts2$molecule <- rep(c("DNA", "RNA"), 20)
nuts2 <- nuts2[order(nuts2$sample, nuts2$year, nuts2$molecule), ]
as.character(nuts2$sample) == as.character(design$Lake)

plot(density(log10(nuts2$DOC)))
plot(density(log10(nuts2$TN)))
plot(density(log10(nuts2$TP)))

# Nutrient Z-Transformation
nuts2$DOC.z <- scale(log(nuts2$DOC))
nuts2$TP.z <- scale(log(nuts2$TP))
nuts2$TN.z <- scale(log1p(nuts2$TN))

beta.dis <- betadisper(vegdist(OTUsREL.log, "bray"), design$Lake)
permutest(beta.dis)

adonis(OTUsREL.log ~ design$Lake + design$Molecule, method = "bray", permutations = 999)
adonis(OTUsREL ~ design$Lake + design$Molecule, method = "bray", permutations = 999)

lake.dbrda <- capscale(OTUsREL.log ~ design$Lake + design$Molecule, distance = "bray", add = T)
anova(lake.dbrda)
RsquareAdj(lake.dbrda)
coef(lake.dbrda)

chem.dbrda <- capscale(OTUsREL.log ~ nuts2$DOC.z + nuts2$TP.z + nuts2$TN.z, 
                       add = T, distance = "bray")
anova(chem.dbrda)
RsquareAdj(chem.dbrda)
coef(chem.dbrda)

anova.cca(chem.dbrda, step=1000)
plot(chem.dbrda)
lmod <- as.mlm(chem.dbrda)
# influence.measures(lmod)
lmod
summary(lmod)
```

## Generalists
```{r}
lake.yr <- paste(design$Lake[design$Molecule == "DNA"], 
                 design$Year[design$Molecule == "DNA"], sep = "")

total <- OTUs[design$Molecule == "DNA", ]
row.names(total) <- lake.yr
totalPA <- (total > 0) * 1

active.rna <- OTUs[design$Molecule == "RNA", ]
row.names(active.rna) <- lake.yr
activePA <- (active.rna > 0) * 1
active <- total * (activePA)
row.names(active) <- lake.yr

activePA.2 <- (active.rna > (total * 0.25)) * 1
active.2 <- total * (activePA.2)
row.names(active.2) <- lake.yr

per <- c(0.001, 0.005, 0.01, 0.05, 0.10)
per.act <- matrix(NA, dim(total)[1], length(per))
per.actT <- matrix(NA, dim(total)[1], length(per))
colnames(per.act) <- per
colnames(per.actT) <- per

  for (i in 1:length(per)){
  activePA.temp <- (active.rna > (total * per[i])) * 1
  active.temp <- total * activePA.temp
  per.act[,i] <- rowSums(active.temp) / rowSums(total)
  per.actT[,i] <- rowSums(activePA.temp) / rowSums(totalPA)
  }

mean.per.act <- colMeans(per.act)
se.per.act <- apply(per.act, 2, se)

mean.per.actT <- colMeans(per.actT)
se.per.actT <- apply(per.actT, 2, se)

par(mar = c(4.5,5,1,1) + 0.1)
plot(mean.per.act ~ per, las = 1, type = "n",
     ylim = c(0.875, 0.99), 
     xaxt = "n", yaxt = "n", xlab = "", ylab = "")

arrows(x0 = per, y0 = mean.per.act, y1 = mean.per.act + se.per.act, angle= 90, length = 0.1, lwd = 2)
arrows(x0 = per, y0 = mean.per.act, y1 = mean.per.act - se.per.act, angle= 90, length = 0.1, lwd = 2)
points(per, mean.per.act, pch = 15, col = "firebrick", cex = 1.25)

axis(side=1, lwd.ticks = 2, tck=-0.02, labels = T, cex.axis = 1)
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1)
axis(side=2, lwd.ticks = 2, labels = T, cex.axis = 1, las = 1, 
     at = c(0.89, 0.92, 0.95, 0.98))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(0.89, 0.92, 0.95, 0.98))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1)
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1)
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(0.89, 0.92, 0.95, 0.98))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(0.89, 0.92, 0.95, 0.98))
mtext(side = 2, "Percent Active", line = 3, cex = 1.5)
mtext(side = 1, "Cutoff", line = 2.5, cex = 1.5)
box(lwd = 2)




plot(mean.per.actT ~ per)

inactive <- total * (1 - activePA)
inactivePA <- (inactive > 0) * 1


gens <- data.frame(matrix(NA, 21, 3))
colnames(gens) <- c("sites", "taxaA", "taxaT")
gens$sites <- c(1:21)

for (i in 1:21){
  gens$taxaA[i] <- sum(colSums(activePA) == i)
  gens$taxaT[i] <- sum(colSums(totalPA) == i)
}

# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)
plot(gens$taxaA ~ gens$sites, xlab = "Number of Sites", ylab = "Number of Taxa")
plot(gens$taxaT ~ gens$sites, xlab = "Number of Sites", ylab = "Number of Taxa")

# Inactive Taxa
inactivePA <- totalPA - activePA
inactive <- total * inactivePA

# Species Abundance Distributions for Active and Inactive Taxa
par(mar = c(4, 5, 1, 1) + 0.1)
plot(sort(log10(colSums(active)), decreasing = T), col = "darkolivegreen3",
     las = 1, xaxt="n", xlab = "", yaxt="n", ylab = "", xlim = c(0, 7000))
points(sort(log10(colSums(inactive)), decreasing = T), col = "cornflowerblue")
axis(side=1, lwd.ticks = 2, tck=-0.02, labels = T, cex.axis = 1, at = c(0, 1500, 3000, 4500, 6000))
axis(side=1, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(0, 1500, 3000, 4500, 6000))
axis(side=2, lwd.ticks = 2, labels = c("10", "1000", "100000"), cex.axis = 1, las = 1, 
     at = c(1, 3, 5))
axis(side=2, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1, 3, 5))
axis(side=3, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(0, 1500, 3000, 4500, 6000))
axis(side=3, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(0, 1500, 3000, 4500, 6000))
axis(side=4, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, at = c(1, 3, 5))
axis(side=4, lwd.ticks = 2, tck=0.01, labels = F, cex.axis = 1, at = c(1, 3, 5))
mtext(side = 2, "Abundance", line = 3.5, cex = 1.5)
mtext(side = 1, "Index", line = 2.5, cex = 1)
legend("topright", c("Inactive", "Active"), fill = c("cornflowerblue", "darkolivegreen3"), bty = "n")
box(lwd = 2)

pro.act <- rowSums(active)/rowSums(total)

rowSums(inactivePA)/rowSums(totalPA)
rowSums(activePA)/rowSums(totalPA)

plot(rowSums(activePA)/rowSums(totalPA), rowSums(inactivePA)/rowSums(totalPA))

inactive <- total * inactivePA
active.N <- total * (1-inactivePA)
active.NPA <- (active.N > 0 ) * 1

plot(rowSums(active.NPA)/rowSums(totalPA), rowSums(inactivePA)/rowSums(totalPA))
plot(rowSums(active.NPA)/rowSums(totalPA) ~ log10(nuts2$TP[nuts2$molecule == "DNA"]),
     xlab = "log10(Total Phosphorus)", ylab = "Proportion of Active Taxa")
phos <- lm(rowSums(active.NPA)/rowSums(totalPA) ~ log10(nuts2$TP[nuts2$molecule == "DNA"]))
abline(phos)
dim(activePA)
dim(totalPA)
```

# Patterns of Resource Diversity
```{r}
# Define Inputs
# Resource = raw site-by-resource matrix
resource.pos <- "../data/SpecAbundAvePos.csv"
resource.neg <- "../data/SpecAbundAveNeg.csv"

# Import Resources
res.in <- read.csv(resource.neg, header=T, row.names=1)

rownames(res.in)
rownames(res.in) <- c("Ann", "blank", "CanyonChemo", "Canyon", "CanyonHypo",
                      "CanyonI", "CanyonII", "CanyonIII", "CanyonIV", "Howe",
                      "Ives", "Jordan", "Lily", "Mountain", "Pony", "Rush",
                      "SecondPine", "UpperPine")

blank <- unlist(res.in["blank", ])
res.hmwf <- res.in[-c(which(rownames(res.in) %in% c("blank", "CanyonChemo", 
                          "CanyonHypo", "CanyonI", "CanyonII",
                          "CanyonIII", "CanyonIV", "Jordan"))), ]
```

## Remove Major Peaks from Blanks
```{r}
summary(blank)
blank[which(blank > 2 * sd(blank))]

# res.hmwf <- res.hmwf[, -c(which(blank > sd(blank)))]

# What other peaks should be removed
for (i in 1:dim(res.hmwf)[1]){
  res.hmwf[i, ] <- res.hmwf[i, ] - blank * 1.1
}

res.hmwf[res.hmwf < 50] <- 0
res.hmwf <- res.hmwf[,colSums(res.hmwf) > 0]
```

## Data Transformations
```{r}
# Remove OTUs with less than two occurences across all sites
res <- res.hmwf

# Sequencing Coverage
coverage <- rowSums(res)

# # Remove Low Coverage Samples (This code removes two sites: Site 5DNA, Site 6cDNA)
# lows <- which(coverage < 10000)
# OTUs <- OTUs[-which(coverage < 10000), ]
# design <- design[-which(coverage < 10000), ]

# Make Relative Abundence Matrices
resREL <- res
for(i in 1:dim(res)[1]){
  resREL[i,] <- res[i,]/sum(res[i,])
}

# Log Transform Relative Resource Abundance
resREL.log <- decostand(resREL, method="log")
```

## Calculate Alpha Diversity
```{r}
# Observed Richness
S.res <- rowSums((res > 0) * 1)

# Simpson's Evenness
res.simpsE <- round(apply(res, 1, SimpE), 3)

# Shannon's Diversity
res.shan <- round(apply(res, 1, H), 2)
res.shan2 <- round(diversity(res, index = "shannon"), 2)

res.div <- as.data.frame(cbind(S.res, res.simpsE, res.shan2))
```

## Figure 5: Resource Diversity
```{r results='hide'}
png(filename="../figures/Figure5.png",
    width = 1600, height = 1200, res = 96*2)
par(opar)
par(mfrow = c(1,1), mar = c(0, 9, 0, 0) + 0.5, oma = c(5, 0, 1, 1) + 0.5)
layout(rbind(1, 2, 3), height = c(3, 3, 3)) 
labs <- c("Ann", "Canyon", "Howe", "Ives", "Lily", "Mountain", "Pony", "Rush",
         "Second\nPine","Upper\nPine")
rich <- barplot(res.div$S.res, names.arg = NULL, las =1, ylim=c(0, 750), 
     xlab = "", ylab = "")
mtext(side = 2, text = "Resource\nRichness", cex.lab = 1.2, line = 4.5)
even <- barplot(res.div$res.simpsE, names.arg = NULL, las =1, ylim=c(0, 0.27), 
     xlab = "", ylab = "")
mtext(side = 2, text = "Simpson's\nEvenness", cex.lab = 1.2, line = 4.5)
shan <- barplot(res.div$res.shan, names.arg = NULL, las = 1, ylim = c(0, 9), 
     xlab = "", ylab = "")
mtext(side = 2, text = "Shannon\nDiversity", cex.lab = 1.2, line = 4.5)
mtext(side = 1, text = labs, line = 2, at = shan, padj = 0.5, cex = 0.8)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)
```

```{r fig.width=8, fig.height=5,echo=FALSE,fig.cap="Resource Alpha Diversity"}
img <- readPNG("../figures/Figure5.png")
grid.raster(img)
```


## Hypothesis that resource diversity is related to nutrient concentration
```{r}
nuts2012 <- nuts2[nuts2$year == "2012" & nuts2$molecule == "DNA", ]
evenmod <- lm(res.div$res.simpsE ~ nuts2012$DOC*nuts2012$TN*nuts2012$TP)
richmod <- lm(res.div$S.res ~ nuts2012$DOC*nuts2012$TN*nuts2012$TP)
summary(evenmod)
summary(richmod)

evenmod <- lm(res.div$res.simpsE ~ log(nuts2012$DOC))
richmod <- lm(res.div$S.res ~ log(nuts2012$DOC))
summary(evenmod)
summary(richmod)

xtable(summary(evenmod))

```

## Hypothesis that resource diversity influences consumer diversity
```{r]}
alpha.div2012 <- alpha.div[alpha.div$Year == "2012" & alpha.div$Molecule == "RNA", c(1, 4:6)]
rownames(alpha.div2012) <- alpha.div2012[, 1]
alpha.div2012 <- alpha.div2012[, -1]
rownames(res.div) == rownames(alpha.div2012)

cor(res.div, alpha.div2012)

rich.mod1 <- lm(alpha.div2012$S.obs ~ res.div$S.res)
rich.mod2 <- lm(alpha.div2012$S.obs ~ res.div$res.simpsE)
summary(rich.mod1)
summary(rich.mod2)
```

```{r}
plot((rowSums(active.NPA)/rowSums(totalPA))[seq(2, 20, by = 2)] ~ res.div$res.simpsE,
     xlab = "Resource Evennes", ylab = "Proportion of Active Taxa")
res.ev <- lm((rowSums(active.NPA)/rowSums(totalPA))[seq(2, 20, by = 2)] ~ res.div$res.simpsE)
abline(res.ev)

```


## Between site comparisions of resources
```{r}
# Calculate Bray-Curtis 
res.db <- vegdist(resREL, method = "bray")

res.pcoa <- cmdscale(res.db, eig = TRUE, k = 3) 
explainvar1 <- round(res.pcoa$eig[1] / sum(res.pcoa$eig), 3) * 100
explainvar2 <- round(res.pcoa$eig[2] / sum(res.pcoa$eig), 3) * 100
explainvar3 <- round(res.pcoa$eig[3] / sum(res.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# Define Plot Parameters
par(mar = c(5, 5, 1, 2) + 0.1)

# Plot Eigenvalues
plot(res.pcoa$eig, xlab = "PCoA Axis", ylab = "Eigenvalue", 
     las = 1, cex.lab = 1.5, pch = 16)

# Add Expectation based on Kaiser-Guttman criterion and Broken Stick Model
abline(h = mean(res.pcoa$eig), lty = 2, lwd = 2, col = "blue")
b.stick <- bstick(10, sum(res.pcoa$eig))
lines(1:10, b.stick, type = "l", lty = 4, lwd = 2, col = "red")

# Add Legend
legend("topright", legend = c("Avg Eigenvalue", "Broken-Stick"), 
       lty = c(2, 4), bty = "n", col = c("blue", "red"))
```


## Figre 6: Resource Differences Across Sites
```{r results='hide'}
png(filename="../figures/Figure6.png",
    width = 1200, height = 1200, res = 96*2)

# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)

# Initiate Plot
plot(res.pcoa$points[ ,1], res.pcoa$points[ ,2], ylim = c(-0.25, 0.3),
     xlim = c(-0.4, 0.4), 
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(res.pcoa$points[ ,1], res.pcoa$points[ ,2],
       pch = 19, cex = 3, bg = "gray", col = "gray")
text(res.pcoa$points[ ,1], res.pcoa$points[ ,2], 
     labels = row.names(res.pcoa$points))
dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)
```

```{r fig.width=8, fig.height=5,echo=FALSE,fig.cap="Resource Ordination"}
img <- readPNG("../figures/Figure6.png")
grid.raster(img)
```



## Resource Explainations of Differences
```{r}

OTUsREL2012.DNA <- OTUsREL[which(design$Year == "2012" & design$Molecule == "DNA"), ]
OTUsREL2012.DNA <- OTUsREL2012.DNA[ , colSums(OTUsREL2012.DNA > 0)]
OTUsREL2012.RNA <- OTUsREL[which(design$Year == "2012" & design$Molecule == "DNA"), ]
OTUsREL2012.RNA <- OTUsREL2012.RNA[ , colSums(OTUsREL2012.RNA > 0)]
active.N2012 <- active.N[grep("2012", rownames(active.N)), ]
rownames(OTUsREL2012.DNA) <- design$Lake[which(design$Year == "2012" & design$Molecule == "DNA")]
rownames(OTUsREL2012.RNA) <- design$Lake[which(design$Year == "2012" & design$Molecule == "DNA")]
rownames(active.N2012) <- design$Lake[which(design$Year == "2012" & design$Molecule == "DNA")]

# DNA
dbrda2012 <- capscale(OTUsREL2012.DNA ~ nuts2012$DOC + nuts2012$TN + nuts2012$TP, add =T, distance = "bray")
anova(dbrda2012)
RsquareAdj(dbrda2012)
coef(dbrda2012)

# RNA
dbrda2012 <- capscale(OTUsREL2012.RNA ~ nuts2012$DOC + nuts2012$TN + nuts2012$TP, add =T, distance = "bray")
anova(dbrda2012)
RsquareAdj(dbrda2012)
coef(dbrda2012)

# Active
dbrda2012 <- capscale(active.N2012 ~ nuts2012$DOC + nuts2012$TN + nuts2012$TP, add =T, distance = "bray")
anova(dbrda2012)
RsquareAdj(dbrda2012)
coef(dbrda2012)

# Resoruces
chem.dbrda <- capscale(resREL ~ nuts2012$DOC + nuts2012$TN + nuts2012$TP, add =T, distance = "bray")
anova(chem.dbrda)
RsquareAdj(chem.dbrda)
coef(chem.dbrda)

anova.cca(chem.dbrda, step=1000)
plot(chem.dbrda)
lmod <- as.mlm(chem.dbrda)
lmod
summary(lmod)

chem.dbrda <- capscale(resREL ~ log(nuts2012$DOC), add =T, distance = "bray")
anova(chem.dbrda)
RsquareAdj(chem.dbrda)
coef(chem.dbrda)

anova.cca(chem.dbrda, step=1000)
plot(chem.dbrda)
lmod <- as.mlm(chem.dbrda)
lmod
summary(lmod)



require(cocorresp)
#test1 <- coca(OTUsREL2012.RNA, resREL, n.axes = 4)

# Variance Partitioning

nuts3 <- cbind(scale(log(nuts2012$DOC)), scale(log1p(nuts2012$TP)))
# nuts3 <- scale(log(nuts2012$DOC))
rownames(nuts3) <- nuts2012$sample

res.db <- vegdist(resREL, method = "altGower")
res.pcoa <- cmdscale(res.db, eig = TRUE, k = 5) 

OTUsREL.log2012 <- OTUsREL.log[design$Year == "2012" & design$Molecule == "RNA", ]
OTUsREL2012 <- OTUsREL[design$Year == "2012" & design$Molecule == "RNA", ]
spe.pcoa <- cmdscale(vegdist(OTUsREL2012, method="bray"), eig = TRUE, k = 5)


# Active
HMWFvarpart <- varpart(spe.pcoa$points[, 1:2], nuts3, res.pcoa$points[, 1:2])
HMWFvarpart <- varpart(OTUsREL.log2012, nuts3, res.pcoa$points)
HMWFvarpart

png(filename="../figures/Figure7.png",
    width = 1200, height = 1200, res = 96*2)

plot(HMWFvarpart)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
par(opar)

anova.cca(rda(spe.pcoa$points, nuts3), step=1000) #[a+b]
anova.cca(rda(spe.pcoa$points, res.pcoa$points), step=1000) #[b+c]
anova.cca(rda(spe.pcoa$points, cbind(nuts3, res.pcoa$points)), step=1000) #[a+b+c]
anova.cca(rda(spe.pcoa$points, res.pcoa$points, nuts3), step=1000) # [a]
anova.cca(rda(spe.pcoa$points, nuts3, res.pcoa$points), step=1000) # [c]

```

# Phylogenetic Approach

Resource distirubiton is not able to explain the distribution of all organisms combined. 
But why should we expect this assumption?

## Remove Cyanobacteria
```{r}




```

# Microbial Functional Groups

Define RDP microbial groups
Test each along with resource differences
Who are the generalist taxa (which are active everywhere)
Are generalists more abundant when resource concentration is higher?

# Can we group resources
What are the similar groups of resources: cluster resources based on abundance
Can we cluster based on chemical data?

```{r}

res.dist <- vegdist(t(resREL), "bray")

res2.pcoa <- cmdscale(res.dist, eig = TRUE, k = 3) 
explainvar1 <- round(res2.pcoa$eig[1] / sum(res2.pcoa$eig), 3) * 100
explainvar2 <- round(res2.pcoa$eig[2] / sum(res2.pcoa$eig), 3) * 100
explainvar3 <- round(res2.pcoa$eig[3] / sum(res2.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)

# Initiate Plot
plot(res2.pcoa$points[ ,1], res2.pcoa$points[ ,2], ylim = c(-0.75, 0.75),
     xlim = c(-0.75, 0.75), 
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(res2.pcoa$points[ ,1], res2.pcoa$points[ ,2],
       pch = 19, cex = 3, bg = "gray", col = "gray")
text(res2.pcoa$points[ ,1], res2.pcoa$points[ ,2], 
     labels = row.names(res.pcoa$points))


# Distances between molecules given sites
summary(resREL) # the data

# Calculate distances
res.dist <- vegdist(resREL, method = "bray")
res.dist2 <- vegdist(t(resREL), method = "bray")

res.pcoa <- cmdscale(res.dist2, eig = TRUE, k = 3) 



# Clustering
res.k <- kmeans(resREL, 5, iter.max = 10)

# plot(resREL, col = res.k$cluster)



# Perform Cluster Analysis
res.ward <- hclust(res.dist, method = "ward.D2")
# Plot Cluster
par(mar = c(1, 5, 2, 2) + 0.1)
plot(res.ward, main = "Sites by Resource")

bac.dist <- vegdist(OTUsREL[design$Year == "2012" & 
                              design$Molecule == "DNA", ], method="bray")
bac.ward <- hclust(bac.dist, method = "ward.D2")
plot(bac.ward, main = "Sites by Species")

```

# Co-Occurance Analysis
```{r}
require("bioDist")

# Subset OTUs for to remove very rare taxa
OTUsREL2 <- OTUsREL[,which(colSums(as.matrix(OTUsREL)) > 0.05)]
resREL2 <- resREL[,which(colSums(as.matrix(resREL)) > 0.022)]

rownames(OTUsREL)
rownames(resREL)
OTUsREL3 <- OTUsREL2[grep("2012_RNA", rownames(OTUsREL)),]
OTUsREL3 <- OTUsREL3[,order(colSums(as.matrix(OTUsREL3)), decreasing = T)[1:100]]
resREL3 <- resREL2[,order(colSums(as.matrix(resREL2)), decreasing = T)[1:100]]
ConRes1 <- cbind(as.matrix(OTUsREL3), as.matrix(resREL3))
ConRes2 <- cbind(as.matrix(resREL3), as.matrix(OTUsREL3))
ConRes <- rbind(ConRes1, ConRes2)

# Calculate 1 - Spearman Correlation Coefficients: Spearman Distance
spear.bac <- spearman.dist(t(OTUsREL2), abs = FALSE)
spear.res <- spearman.dist(t(as.matrix(resREL2)), abs = FALSE)
spear.ConRes <- spearman.dist(t(as.matrix(ConRes)), abs = FALSE, diag = F, upper = F)

spear.bac2 <- spear.bac - 1
spear.bac3 <- spear.bac2
spear.bac3[which(spear.bac2 < 0.5 & spear.bac2 > -0.5)] <- NA

spear.res2 <- spear.res - 1
spear.res3 <- spear.res2
spear.res3[which(spear.res2 < 0.5 & spear.res2 > -0.5)] <- NA

spear.ConRes2 <- spear.ConRes - 1
spear.ConRes3 <- spear.ConRes2
spear.ConRes3[which(spear.ConRes2 < 0.5 & spear.ConRes2 > -0.5)] <- NA
spear.ConRes4 <- as.matrix(spear.ConRes3) # [1:100, 101:200]

# Custome Color Palette
jet.colors <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan", 
                                 "#7FFF7F", "yellow", "#FF7F00", "red", 
                                 "#7F0000"))

layout(rbind(c(1,2)))

require(gridExtra)
order <- rev(attr(spear.bac3, "Labels"))
plot1 <- levelplot(as.matrix(spear.bac3)[,order], aspect = "iso", col.regions = jet.colors, 
          xlab = list(label = "OTU", cex = 1.2), ylab = list(label = "OTU", cex = 1.2), 
          scales = list(cex = 0.3, x = list(rot=90)),
          main = "Species Co-Occurence")


order <- rev(attr(spear.res3, "Labels"))
plot2 <- levelplot(as.matrix(spear.res3)[,order], aspect = "iso", col.regions = jet.colors, 
          xlab = list(label = "Resource", cex = 1.2), ylab = list(label = "Resource", cex = 1.2), 
          scales = list(cex = 0.3, x = list(rot=90)),
          main = "Resource Co-Occurence")
plot1
plot2

grid.arrange(plot1, plot2, ncol = 2)


plot3 <- levelplot(spear.ConRes4, aspect = "iso", col.regions = jet.colors, 
          xlab = list(label = "Consumer", cex = 1.5), 
          ylab = list(label = "Resource", cex = 1.5), 
          scales = list(cex = 0.3, tck = 0.75,
                        #x = list(labels=colnames(ConRes)[1:100], rot=90),
                        #y = list(labels=colnames(ConRes)[101:200])),
                        x = list(rot=90),
                        y = list()),
          main = list(label = "Consumer-Resource Co-Occurence", line = 1),
          par.settings=list(layout.heights=list(top.padding=3, bottom.padding=3)))         
plot3
             
             
             
spearB.pcoa <- cmdscale(spear.bac, eig = TRUE, k = 3) 
explainvar1 <- round(spearB.pcoa$eig[1] / sum(spearB.pcoa$eig), 3) * 100
explainvar2 <- round(spearB.pcoa$eig[2] / sum(spearB.pcoa$eig), 3) * 100
explainvar3 <- round(spearB.pcoa$eig[3] / sum(spearB.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

par(mar = c(5, 5, 1, 1) + 0.1)

# Define Plot Symbols
lake.pch <- rep(NA, length(design$Molecule))
for (i in 1:length(design$Molecule)){
if (design$Molecule[i] == "DNA"){
    lake.pch[i] <- 16
  }else{
    lake.pch[i] <- 17
  }}


# Initiate Plot
plot(spearB.pcoa$points[ ,1], spearB.pcoa$points[ ,2], 
     ylim = c(-0.4, 0.8), xlim = c(-0.5, 0.6), 
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = lake.pch, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, 
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(spearB.pcoa$points[ ,1], spearB.pcoa$points[ ,2],
       pch = lake.pch, cex = 0.5, bg = "gray", col = "gray")

ordiellipse(cbind(spearB.pcoa$points[ ,1], spearB.pcoa$points[ ,2]),
    design$Lake, kind="sd", conf=0.95,
    lwd=2, draw = "polygon", col="gray", border = "black", label=TRUE, 
    cex=1, bty = 'n')


```